### ISSUE TITLE
--rbss Trim Galore version: 0.4.3 not removing two extra basepairs

### ISSUE THREAD

**Issue by bri2020** [FelixKrueger/TrimGalore#8]
Dear @Felix,
I am using your nice tool trim galore and the --rrbs function on my RRBS data. 
I checked my fastq before and after trimming and I do not understand, why no 2 extra bps are trimmed at 3'end. Here I am copying an example read, where only the adaptor is cut, but not the two extra base pairs. Initially I thought that this did not happen because of quality trimming, but that is not the case. Could you have a look. Maybe I am missing something really basic??
Thanks a lot and best wishes!

RAW
```

@header [I changed this for posting]
TGGGTAGGTGTGAGAAGTGATTTGTTTGTAGGTGGAAGATAGAAATTTTGGAAAGTGGAAAGTTTGAGATCGGAAGAGCACACGTCTGAACTCCAGTCACA
+
BBCCBGGGGGGGFG?FGGGGGGG>GGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGG1FGGGGGGGGGFFGGGGGGGGEFGGG<EF>EFGEE?F;?>EFF0
```
TRIMMED
```
@header [I changed this for posting]
TGGGTAGGTGTGAGAAGTGATTTGTTTGTAGGTGGAAGATAGAAATTTTGGAAAGTGGAAAGTTTG
+
BBCCBGGGGGGGFG?FGGGGGGG>GGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGG1FGGGGGGG
```

Here are some extractions from the report
"Trimming mode: single-end
**Trim Galore version: 0.4.3
Cutadapt version: 1.9.1**
Quality Phred score cutoff: 20
Quality encoding type selected: ASCII+33
Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)
Maximum trimming error rate: 0.1 (default)
Minimum required adapter overlap (stringency): 1 bp
Minimum required sequence length before a sequence gets removed: 20 bp
**File was specified to be an MspI-digested RRBS sample. Read 1 sequences with adapter contamination will be trimmed a further 2 bp from their 3' end**, and Read 2 sequences will be trimmed by 2 bp from their 5' end to remove potential methylation-biased bases from the end-repair reaction
Running FastQC on the data once trimming has completed
Output file will be GZIP compressed
This is cutadapt 1.9.1 with Python 2.7.6
Command line parameters: -f fastq -e 0.1 -O 1 -a AGATCGGAAGAGC F12477-L2_S4_L001_R1_001.fastq.gz_qual_trimmed.fastq
Trimming 1 adapter with at most 10.0% errors in single-end mode ...
Finished in 174.08 s (19 us/read; 3.19 M reads/minute).

=== Summary ===

Total reads processed:               9,241,550
Reads with adapters:                 3,703,585 (40.1%)
Reads written (passing filters):     9,241,550 (100.0%)

Total basepairs processed:   911,214,356 bp
Total written (filtered):    866,777,499 bp (95.1%)

=== Adapter 1 ===

Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3703585 times.

No. of allowed errors:
0-9 bp: 0; 10-13 bp: 1

Bases preceding removed adapters:
  A: 27.0%
  C: 1.7%
  G: 37.0%
  T: 26.6%
  none/other: 7.6%

9241550 sequences processed in total
Sequences were truncated to a varying degree because of deteriorating qualities (Phred score quality cutoff: 20):	2906090 (31.4%)
Sequences removed because they became shorter than the length cutoff of 20 bp:	299958 (3.2%)
**RRBS reads trimmed by additional 2 bp when adapter contamination was detected:	0 (0.0%)**

**Comment by FelixKrueger** [comment]
Thanks for reporting this, something seems to have crept in between releases 0.4.1 and 0.4.2 that leads to this issue. Will report back as soon as I found the reason.

**Comment by FelixKrueger** [comment]
I have now found and fixed the issue, apparently the trimming was only working as intended for paired-end files... Many thanks for bringing this to my attention! You should be able to clone the latest version and find it all in a working state. Fixed in 766274c5fb879806e0ef2a7eba1d5c5fd378c604.

**Comment by bri2020** [comment]
Thanks!
